back-end/src/sciler/config/workingConfigTypes.go

```diff
@@ -0,0 +1,320 @@
+package config
+
+import (
+	"fmt"
+	"reflect"
+)
+
+// WorkingConfig has additional fields to ReadConfig, with lists of conditions, constraints and actions.
+type WorkingConfig struct {
+	General       General
+	Puzzles       []Puzzle
+	GeneralEvents []GeneralEvent
+	Devices       map[string]Device
+}
+
+// Rule is a struct that describes how action flow is handled in the escape room.
+type Rule struct {
+	ID          string
+	Description string
+	Limit       int
+	Conditions  LogicalCondition
+	Actions     []Action
+}
+
+// Puzzle is a struct that describes contents of a puzzle.
+type Puzzle struct {
+	Event GeneralEvent
+	Hints []string
+}
+
+// GetName returns the name of a Puzzle
+func (p Puzzle) GetName() string {
+	return p.Event.Name
+}
+
+// GetRules returns the rules of a Puzzle
+func (p Puzzle) GetRules() []Rule {
+	return p.Event.Rules
+}
+
+// GeneralEvent defines a general event, like start.
+type GeneralEvent struct {
+	Name  string `json:"name"`
+	Rules []Rule `json:"rules"`
+}
+
+// GetName returns the name of a GeneralEvent
+func (g GeneralEvent) GetName() string {
+	return g.Name
+}
+
+// GetRules returns the rules of a GeneralEvent
+func (g GeneralEvent) GetRules() []Rule {
+	return g.Rules
+}
+
+// Event is an interface that both Puzzle and GeneralEvent implement
+type Event interface {
+	GetName() string
+	GetRules() []Rule
+}
+
+func compare(param1 interface{}, param2 interface{}, comparision string) bool {
+	switch comparision {
+	case "eq":
+		return reflect.DeepEqual(param1, param2)
+	case "lt":
+		return param1.(float64) < param2.(float64)
+	case "gt":
+		return param1.(float64) > param2.(float64)
+	case "lte":
+		return param1.(float64) <= param2.(float64)
+	case "gte":
+		return param1.(float64) >= param2.(float64)
+	case "contains":
+		return contains(param1, param2)
+	default:
+		panic(fmt.Sprintf("Cannot compare on: %s", comparision))
```

Comment:
shouldn't this just be logged? also, error messages in lower case

Reply:
indeed lowercase, and no this should panic, but this should never happen since all types and comparisons are already checked when reading in JSON